File: 0075-sort-colors/0075-sort-colors.cpp
class Solution
{
    public:
        void sortColors(vector<int> &nums)
        {
            int z = 0, t = 0;
            for (int x: nums)
            {
                if (x == 0)
                {
                    ++z;
                }
                if (x == 2)
                {
                    ++t;
                }
            }
            for (int i = 0; i < nums.size(); i++)
            {
                nums[i] = 1;
            }
            int i = 0, j = nums.size() - 1;
            while (z--)
            {
                nums[i++] = 0;
            }
            while (t--)
            {
                nums[j--] = 2;
            }
        }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int zero = 0, two = 0;
        
        for(int i = 0; i < nums.length; i++){
            zero += (nums[i] == 0 ? 1 : 0);
            two += (nums[i] == 2 ? 1 : 0);
            
            nums[i] = 1;
        }
        
        int i = 0;
        
        while(i < nums.length && zero > 0){
              nums[i++] = 0;
            --zero;
        }
        
        i = nums.length - 1;
        
        while(i >= 0 && two > 0){
            nums[i--] = 2;
            --two;
        }
    }
}